### ISSUE TITLE
Do SWE Bench Lite benchmark results use the hints_text?

### ISSUE THREAD

**Issue by paul-gauthier** [OpenDevin/OpenDevin#2149]
I was looking at your [SWE Bench Lite results on huggingface](https://huggingface.co/spaces/OpenDevin/evaluation) and noticed that some of the versions are named `1.5-nohint`. I also see that the [inference code uses hints based on the `USE_HINT_TEXT` environment variable](https://github.com/OpenDevin/OpenDevin/blob/f4bc52461a3b5d8c280aa18040df0c414ddc1b8b/evaluation/swe_bench/run_infer.py#L27).

Does this mean that the rest of the results being reported are using the SWE Bench `hints_text` field to solve the problems?

**Comment by paul-gauthier** [comment]
The discussion in #2136 makes me think this is the case, and that the Open Devin results used the hints unless marked otherwise.

What is the best result from Open Devin without hints? Is it the 16.67% from this row of the huggingface page?

```
|CodeActSWEAgent | v1.5-no-hint | gpt-4o-2024-05-13 | 16.67 
```

**Comment by li-boxuan** [comment]
> What is the best result from Open Devin without hints? Is it the 16.67% from this row of the huggingface page?

I believe so. @xingyaoww might be running more benchmarks though...

**Comment by xingyaoww** [comment]
Yes - we currently default to use "hint" for eval. This was an unfortunate bug we introduced in the early version of eval harness, and we want to be completely honest & transparent about it. We only realized this when one of our contributors pointed out the code with the "with hint" issue a few days ago.

We are still running more benchmarks (w/ and w/o hint) to confirm. Still, the general conclusion at this moment is that, without hints and only running on each problem *once* - the agent's performance is similar around the level of swe-agent and ACR. That performance is slightly worse for general agent that can do browsing & GitHub (more potential source of confusion in the prompt).

We plan to release a paper/report in the coming weeks to discuss more about the results -- stay tuned! :)
